spec/support/login_macros.rb

```diff
@@ -7,5 +7,11 @@ def fake_login
     allow(controller).to receive(:current_user).and_return(@current_user)
     allow(controller).to receive(:logout_if_not_user_credentials).and_return(nil)
   end
-
+  
+  def fake_login_known_user(user)
```

Comment:
Assignment Branch Condition size for fake_login_known_user is too high. [15.03/15]